Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java

```diff
@@ -119,12 +119,12 @@ public boolean equals(Object obj) {
   public static AnalysisError fromJson(JsonObject jsonObject) {
     String severity = jsonObject.get("severity").getAsString();
     String type = jsonObject.get("type").getAsString();
-    String code = jsonObject.get("code") == null ? null : jsonObject.get("code").getAsString();
     Location location = Location.fromJson(jsonObject.get("location").getAsJsonObject());
     String message = jsonObject.get("message").getAsString();
     String correction = jsonObject.get("correction") == null ? null : jsonObject.get("correction").getAsString();
+    String code = jsonObject.get("code").getAsString();
```

Comment:
I suppose that in some sense we could mark all fields returned by the server as being optional, but I think it's useful to the implementors of clients to know which fields they can count on being present, so I'd rather not change it in the spec.

That leaves us with three options that I can think of:

1. Decide that it isn't really a problem that we can't use an SDK older than 1.16, given that we're about to ship 1.19.

2. Manually change the copy of the generated Java to tolerate missing data. I don't know how frequently the API interface classes are updated, so I don't know how often this would need to be done before we stopped caring about 1.16.

3. Update the generator to make the code always be backward compatible by always tolerating missing data.